### PR TITLE
chore: Update contributing testing documentation

### PR DESCRIPTION
**Pull request by binarylogic**
While reviewing https://github.com/timberio/vector/pull/3634/files, I noticed we don't elaborate on how to unit and integration test, so I decided to add it here, among other testing aspects.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -190,6 +207,15 @@ $ curl -u "$GITHUB_USERNAME:$GITHUB_TOKEN" \
   --data '{"ref": "$GIT_REF}'
 ```
 
+##### Test harness
+
+You can invoke the [test hardness][urls.vector_test_harness] by commenting on
```
Comment (jamtur01) [inline CONTRIBUTING.md:212]: harness. :)

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -429,25 +455,55 @@ to detect common problems.
...
-This will create a `100MiB` sample log file in the `sample.log` file.
+#### Integration Tests
+
+Integration tests verify that Vector actually works with the services it
+integrates with. Unlike unit tests, integration tests require external services
+to run. A few rules when setting up integration tests:
+
+- [ ] To ensure all contributors can run integration tests, the service must
+      run in a Docker container.
+- [ ] The service must be configured on a unique port that configured through an
```
Comment (jamtur01) [inline CONTRIBUTING.md:480]: The service must be configured on a unique port that *is* configured through an

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -429,25 +455,55 @@ to detect common problems.
...
+      environment variable.
+- [ ] Add a `test-integration-<name>` to Vector's [`Makefile`](/Makefile) and
```
Comment (jamtur01) [inline CONTRIBUTING.md:482]: that it starts the service...

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -429,25 +455,55 @@ to detect common problems.
...
+Once complete, you can run your integration tests with:
+
+```bash
+make test-integration-<name>
+```
+
+#### Blackbox Tests
+
+Vector also offers blackbox testing via
+[Vector's test hardness][urls.vector_test_harness]. This is a complex testing
```
Comment (jamtur01) [inline CONTRIBUTING.md:497]: harness

**Review by jamtur01 — CHANGES_REQUESTED** [review]
Comments inline.

File: CONTRIBUTING.md (hunk 1 above)
Comment (binarylogic) [reply CONTRIBUTING.md:212]: Ha ha, fixed.

**Review by jszwedko — APPROVED** [review]
Lost track of this one, looks good!

**Comment by jszwedko** [comment]
Or I guess I didn't lose track, I was looking at the PR open time rather than review request time :smile:

**Pull request merged**